### ISSUE TITLE
"Canceled desc = context canceled" without any explanations

### ISSUE THREAD

**Issue by markeloff86** [tinode/chat#871]
## Subject of the issue
"Canceled desc = context canceled" without any explanations

## Describe your issue here.
I try to integrate my python app with a Tinode chat instance.
I'm using gRPC library for python 0.22.3 version.
When I sent a request from my app using root user like this:
```python
def subscribe_topic_message(message_id, user_id, query, on_behalf_of=None):
    client_sub = pb.ClientSub(id=str(message_id), topic=user_id, get_query=pb.GetQuery(what=query))
    
    if on_behalf_of:
        return pb.ClientMsg(sub=client_sub, on_behalf_of=on_behalf_of)
    
    return pb.ClientMsg(sub=client_sub)
```
If on_behalf_of isn't None then I get "gRPC failed with StatusCode.UNKNOWN: Exception iterating requests!" on my client side.
Meanwhile on the server side I see only this log without any explanation what the problem was "E2023/06/14 06:09:35 grpc: recv utETkAcEWFc rpc error: code = Canceled desc = context canceled"

#### Server-side
- [ ] Your own setup:
  * Linux
  * tinode/tinode-mongodb:0.22
  * mongo:4.4.18
  * standalone

#### Client-side
- [ ] Your own client. Describe it:
  * gRPC
  * Python
  * grpcio==1.54.2

### Expected behaviour
More logs about the cause of the error 

### Actual behaviour
Poor logs about error

### Server-side log
I2023/06/14 06:08:36 grpc: session started utETkAcEWFc 184.22.3.234:51171 2
I2023/06/14 06:08:36 grpc in: hi:{id:"4039284" user_agent:"domcom-backend/1.0.0 (Darwin/22.5.0); gRPC-python/0.22.3+1.54.2" ver:"0.22.3" lang:"EN"} utETkAcEWFc
I2023/06/14 06:08:36 grpc in: login:{id:"4039285" scheme:"basic" secret:"root:domcomQWErty1!"} utETkAcEWFc
I2023/06/14 06:09:32 grpc in: acc:{id:"340833" user_id:"new" scheme:"basic" secret:"79139638082:test12345!" desc:{public:"{\"fn\": \"\\u0416\\u0438\\u043b\\u0435\\u0446 \\u0422\\u0435\\u0441\\u0442\\u043e\\u0432\\u044b\\u0439\"}"}} utETkAcEWFc
I2023/06/14 06:09:33 grpc in: sub:{id:"3210414" topic:"grp6jKn7OmIVj8" get_query:{}} utETkAcEWFc
I2023/06/14 06:09:34 grpc in: set:{id:"3210415" topic:"grp6jKn7OmIVj8" query:{sub:{user_id:"usrY0o2ObbwVJ4"}}} utETkAcEWFc
I2023/06/14 06:09:34 grpc in: sub:{id:"408828" topic:"grpUXJgXp_28e4" get_query:{}} utETkAcEWFc
I2023/06/14 06:09:35 grpc in: set:{id:"408829" topic:"grpUXJgXp_28e4" query:{sub:{user_id:"usrY0o2ObbwVJ4"}}} utETkAcEWFc
E2023/06/14 06:09:35 grpc: recv utETkAcEWFc rpc error: code = Canceled desc = context canceled

### Client-side log
[14/Jun/2023 06:12:11,849] TINODE API:gRPC failed with StatusCode.UNKNOWN: Exception iterating requests!

**Comment by markeloff86** [comment]
P.S this code perfectly worked with 0.16.7 tinode and gRPC lib version
The problem appeared only after updating.

**Comment by or-else** [comment]
Yes, unfortunately python gRPC is brittle and not debugging it is not easy. I'm pretty sure this is not a bug in Tinode.

`Canceled desc = context canceled` means connection is lost.

The message `gRPC failed with StatusCode.UNKNOWN: Exception iterating requests!` means you message generator (the one with the `yield`) has thrown an exception.  Wrap its content with `try ... except` and see what's throwing.

Please direct further questions to the forum.

**Comment by or-else** [comment]
And if you want better debugging logs then please file a bug with Google: https://github.com/grpc/grpc/issues
